src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -1363,6 +1379,60 @@ static boolean isLegalName(String name) {
+        FileProperties currentProperties = getFileProperties();
+        FileMetadata currentMetadata = currentProperties.fileMetadata.orElse(new FileMetadata());
+        FileMetadata newMetadata = currentMetadata.withCreationDate(newCreationDate);
+
+        setModified();
+
+        RetrievedCapability ourPointer = linkPointer.orElse(pointer);
+        WritableAbsoluteCapability us = (WritableAbsoluteCapability) ourPointer.capability;
+        CryptreeNode nodeToUpdate = ourPointer.fileAccess;
+
+        boolean isDir = this.isDirectory();
+        boolean isLink = ourPointer.getProperties().isLink;
+
+        FileProperties newProperties = new FileProperties(currentProperties.name, isDir, isLink,
+                currentProperties.mimeType, currentProperties.size,
+                currentProperties.modified, currentProperties.isHidden,
+                currentProperties.thumbnail, currentProperties.streamSecret, newMetadata);
+
+        SigningPrivateKeyAndPublicHash signer = isLink ? parent.signingPair() : signingPair();
+        return userContext.network.synchronizer.applyComplexUpdate(owner(), signer,
+                (s, committer) -> nodeToUpdate.updateProperties(s, committer, us,
+                        entryWriter, newProperties, userContext.network)
+        ).thenCompose(newVersion -> parent.getUpdated(newVersion, userContext.network));
+    }
+
+    @JsMethod
+    public CompletableFuture<FileWrapper> setGpsCoordinates(FileWrapper parent, Coordinates newGpsData, UserContext userContext) {
+        FileProperties currentProperties = getFileProperties();
+        FileMetadata currentMetadata = currentProperties.fileMetadata.orElse(new FileMetadata());
+        FileMetadata newMetadata = currentMetadata.withGpsData(newGpsData);
+
+        setModified();
+
+        RetrievedCapability ourPointer = linkPointer.orElse(pointer);
+        WritableAbsoluteCapability us = (WritableAbsoluteCapability) ourPointer.capability;
+        CryptreeNode nodeToUpdate = ourPointer.fileAccess;
+
+        boolean isDir = this.isDirectory();
+        boolean isLink = ourPointer.getProperties().isLink;
+
+        FileProperties newProperties = new FileProperties(currentProperties.name, isDir, isLink,
+                currentProperties.mimeType, currentProperties.size,
+                currentProperties.modified, currentProperties.isHidden,
+                currentProperties.thumbnail, currentProperties.streamSecret, newMetadata);
+
+        SigningPrivateKeyAndPublicHash signer = isLink ? parent.signingPair() : signingPair();
+        return userContext.network.synchronizer.applyComplexUpdate(owner(), signer,
+                (s, committer) -> nodeToUpdate.updateProperties(s, committer, us,
+                        entryWriter, newProperties, userContext.network)
+        ).thenCompose(newVersion -> parent.getUpdated(newVersion, userContext.network));
+    }
+
```

Comment:
🟢 
Just wondering - those methods looks almost the same, seems to me that the only difference is the line `FileMetadata newMetadata = currentMetadata.withGpsData(newGpsData);`. Maybe we could create a new method/class and reuse it to avoid DRY?

Reply:
Done :)